api/clients/v2/verification/errors.go

```diff
@@ -45,3 +48,33 @@ type CertVerifierInvalidCertError struct {
 func (e *CertVerifierInvalidCertError) Error() string {
 	return fmt.Sprintf("invalid cert: call to CertVerifier failed with status code %d: %s", e.StatusCode, e.Msg)
 }
+
+// IsEVMOutOfGasError tries detecting an EVM out-of-gas error. This assumes the RPC provider
+// returns a standard error message and does not malform the EVM return data in transit.
+func IsEVMOutOfGasError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// Try to catch go-ethereum's RPC error (might be wrapped or modified based on provider)
+	var rerr rpc.Error
+	if errors.As(err, &rerr) {
+		msg := strings.ToLower(rerr.Error())
```

Comment:
wait this only works with go-ethereum's RPC? Then this is probably not an acceptable solution?

Reply:
what would an alternative solution look like? for clarification `go-ethereum` might not be the best term here but there is a fallback conditional check with asserts for substrings defined in node execution envs